ydb/library/yql/parser/pg_wrapper/postgresql/src/backend/utils/adt/timestamp.c

```diff
@@ -5750,6 +5750,16 @@ generate_series_timestamp(PG_FUNCTION_ARGS)
 		Interval   *step = PG_GETARG_INTERVAL_P(2);
 		MemoryContext oldcontext;
 		Interval	interval_zero;
+		int step_sign;
+
+		/* Determine sign of the interval */
```

Comment:
pg sources should not be patched without a good reason

Reply:
I believe its an inefficiency in PG. There are examples of early exits, e.g. https://a.yandex-team.ru/arcadia/contrib/ydb/library/yql/parser/pg_wrapper/postgresql/src/backend/utils/adt/numeric.c?rev=r13318364#L1599